service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/UploadC2DocumentsController.java

```diff
@@ -49,13 +54,23 @@ public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(
         @RequestHeader(value = "authorization") String authorization) {
         Map<String, Object> data = callbackrequest.getCaseDetails().getData();
         CaseData caseData = mapper.convertValue(data, CaseData.class);
-        
+
         data.put("c2DocumentBundle", buildC2DocumentBundle(caseData, authorization));
         data.remove("temporaryC2Document");
 
         return AboutToStartOrSubmitCallbackResponse.builder().data(data).build();
     }
 
+    @PostMapping("/submitted")
+    public void handleSubmittedEvent(
+        @RequestHeader(value = "authorization") String authorization,
+        @RequestHeader(value = "user-id") String userId,
+        @RequestBody CallbackRequest callbackRequest) {
+
+        applicationEventPublisher.publishEvent(new C2UploadedEvent(callbackRequest, authorization, userId,
+            userDetailsService.getUserDetails(authorization)));
```

Comment:
let's just get userDetails in the eventHandler rather than passing it through? We need to set it in the handler anyway so 🤷‍♂

Reply:
yes we could. Will update